_models/models.js

```diff
@@ -7,17 +8,40 @@ exports.selectCategories = () => {
     })
 }
 
-exports.selectReviewsWithComCounts = () => {
+exports.checkQueryParams = (category, sort_by, order) => {
     return db.query(`
-                    SELECT reviews.review_id, reviews.owner, reviews.title, reviews.category, reviews.review_img_url, reviews.created_at, reviews.votes, reviews.designer,  COUNT(comments.body) AS comment_count
-                    FROM comments
-                    RIGHT JOIN reviews
-                    ON comments.review_id = reviews.review_id
-                    GROUP BY reviews.review_id
-                    ORDER BY reviews.created_at DESC;
                     `)
-        .then(({rows})=>{
-            // console.log(">>> selected reviews >>>", rows)
+        .then(({rows}))
+}
+
+exports.selectReviewsWithComCounts = (category = 'social deduction', sort_by = 'created_at', order = 'DESC' ) => {
+
+    if (!['dexterity', "children's games", "social deduction", "euro game"].includes(category)) {
+        return Promise.reject({status: 400, msg: 'Bad Request: Invalid category '})
+    }
+    if (!['ASC','DESC'].includes(order)) {
+        return Promise.reject({status: 400, msg: 'Bad Request: Invalid order query'})
+    }
+    if (!['review_id','owner','title','category','created_at','votes','designer','comment_count','review_img)url'].includes(sort_by)) {
+        return Promise.reject({status: 400, msg: 'Bad Request: Invalid sort query'})
+    }
+
+    let query = `
+                SELECT reviews.review_id, reviews.owner, reviews.title, reviews.category, reviews.review_img_url, reviews.created_at, reviews.votes, reviews.designer,  COUNT(comments.body) AS comment_count
+                FROM comments
+                RIGHT JOIN reviews
+                ON comments.review_id = reviews.review_id
+                WHERE category = $1
+                GROUP BY reviews.review_id
+                ORDER BY reviews.${sort_by} ${order};
+                `;
```

Comment:
We should use $2 and $3 instead of sort_by and order here, to avoid SQL injection if these are invalid.

Reply:
Please use what John mentioend above